source/common/json/json_internal.cc

```diff
@@ -383,6 +384,27 @@ nlohmann::json Field::asJsonDocument() const {
 
 uint64_t Field::hash() const { return HashUtil::xxHash64(asJsonString()); }
 
+ValueType Field::getValue(const std::string& name) const {
+  auto value_itr = value_.object_value_.find(name);
+  if (value_itr == value_.object_value_.end()) {
+    throw Exception(fmt::format("key '{}' missing from lines {}-{}", name, line_number_start_,
+                                line_number_end_));
+  }
+  switch (value_itr->second->type_) {
+  case Type::Boolean:
+    return value_itr->second->booleanValue();
+  case Type::Double:
+    return value_itr->second->doubleValue();
+  case Type::Integer:
+    return value_itr->second->integerValue();
+  case Type::String:
+    return value_itr->second->stringValue();
+  default:
+    throw Exception(fmt::format("key '{}' not a value type from lines {}-{}", name,
```

Comment:
Will this be called on data plane path or control/config path?

exception is disallowed on the data plane. https://github.com/envoyproxy/envoy/blob/main/STYLE.md#deviations-from-google-c-style-guidelines

Reply:
Hello, unfortunately it's in data plane for json to metadata filter while we have an unexpected json.
Actually json util has exceptions here and there tho but we should stop adding more exception.

Should we consider to redefine 
`using ValueType = absl::variant<bool, int64_t, double, std::string, absl::monostate>;` where `absl::monostate` indicates parse error?

Reply:
Actually it's OK in a sense, and in any case is not lint-checked to *throw* on the data plane, but it's invalid in Envoy to *catch* on the data plane, and that should be lint-checked. So this will become an unhandled exception.

Sorry -- I merged this since it was approved but did not review it.

Reply:
do we need. to roll this back?

Reply:
the json to metadata is not landed and this method is not used so a fix should be enough

Reply:
Yea,  good to know.
Regarding the fix: I think absl::monostate usually represent empty state, you can also use log OR use absl::StatusOr to return error status if you want further processing on the caller site?

Reply:
absl::StatusOr would be a good way to indicate an error from this function, IMO.

Reply:
`json_loader_lib` is pretty isolated so not depending on log module at the moment. i'll do `StatusOr`

Reply:
With that being said, if we want to merge json_to_metadata filter, several json utils need NoThrow version

Reply:
Agreed. That will become obvious during CI if you otherwise need to have a `catch`.